_posts/2021-04-21-security-incident-disclosure.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Security Incident Disclosure
+author: reitermarkus
+---
+
+On 18th April 2021 [a security researcher identified](https://blog.ryotak.me/post/homebrew-security-incident-en/) a vulnerability in our `review-cask-pr` GitHub Action used to automatically review and approve cask version bump pull requests which was automatically merged by our `automerge` GitHub Action. This was reported via HackerOne and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) was submitted with our permission. The [pull request was subsequently reverted](https://github.com/Homebrew/homebrew-cask/pull/104197), the [`automerge` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104199) and the [`review-cask-pr` action disabled and removed](https://github.com/Homebrew/homebrew-cask/pull/104204) from all the `homebrew-cask*` repositories that were using it.
```

Comment:
```suggestion
On 18th April 2021, [a security researcher identified](https://blog.ryotak.me/post/homebrew-security-incident-en/) a vulnerability in our `review-cask-pr` GitHub Action. The vulnerability affected:

1. the `homebrew-cask` repository and
2. all `homebrew-cask-*` repositories (taps) on the Homebrew organization, except those for which we never enabled the Action.

Whenever an affected cask repository received a pull request which just changed the version of a cask and nothing else, the `review-cask-pr` GitHub Action would automatically review and approve the pull request. The approval would then trigger another GitHub Action named `automerge`, which in turn would pick up the approved pull request and merge it. This was reported via HackerOne and [a pull request demonstrating the vulnerability](https://github.com/Homebrew/homebrew-cask/pull/104191) (PoC) was submitted with our permission. We subsequently [reverted the PoC pull request](https://github.com/Homebrew/homebrew-cask/pull/104197), [disabled and removed the `automerge` Action](https://github.com/Homebrew/homebrew-cask/pull/104199) and [disabled and removed the `review-cask-pr` Action](https://github.com/Homebrew/homebrew-cask/pull/104204) from the affected cask repositories.
```

Summary of my changes:

1. use simpler language to make the text easier to read;
2. clarify that the Actions were written by us, not by GitHub;
3. clarify that `homebrew-cask` was specifically affected;
4. clarify that a number of taps were not affected (I don’t know which ones though);
5. clarify what a cask version bump is;
6. clarify which one of the pull requests mentioned was reverted (i. e., the PoC one);
7. use the active voice more often;
8. insert a missing comma;
9. capitalize the word _Action_ consistently.

Reply:
Sounds good, a few quibbles:

> * clarify that a number of taps were not affected (I don’t know which ones though);

All homebrew-cask* taps were technically affected by the vulnerability and brew/homebrew-core/external command taps were not.

> 9\. capitalize the word _Action_ consistently.

I think we should use "GitHub Action" or "GitHub Actions" but "action" or "actions" (without the GitHub prefix).

> 6\. (i. e., the PoC one);

Maybe worth making this "proof of concept" for those unfamiliar? 

@claui any chance you could amend this accordingly?

Reply:
Actually, ignore me, I'll apply this then edit it.